sdk/azidentity/doc.go

```diff
@@ -0,0 +1,89 @@
+// +build go1.13
+
+// Copyright 2017 Microsoft Corporation. All rights reserved.
+// Use of this source code is governed by an MIT
+// license that can be found in the LICENSE file.
+
+/*
+Package azidentity implements a set of credential types for use with
+Azure SDK clients that support Azure Active Directory (AAD) token authentication.
+
+The following credential types are included in this module:
+	- AuthorizationCodeCredential
+	- AzureCLICredential
+	- ChainedTokenCredential
+	- ClientCertificateCredential
+	- ClientSecretCredential
+	- DefaultAzureCredential
+	- DeviceCodeCredential
+	- EnvironmentCredential
+	- InteractiveBrowserCredential
+	- ManagedIdentityCredential
+	- UsernamePasswordCredential
+
+By default, the recommendation is that users call NewDefaultAzureCredential() which will
+provide a default ChainedTokenCredential configuration conformed of:
+	- EnvironmentCredential
+	- ManagedIdentityCredential
+	- AzureCLICredential
+Configuration options can be used to exclude any of the previous credentials from the
+DefaultAzureCredential implementation.
+
+Example call to NewDefaultAzureCredential():
+	cred, err := NewDefaultAzureCredential(nil) // pass in nil to get default behavior for this credential
+	if err != nil {
+		// process error
+	}
+	client := armresources.NewResourcesClient(armcore.NewDefaultConnection(cred, nil), "<subscription ID>")
```

Comment:
I wonder if it would be better to simply say something like `// use cred ....` instead of something this specific.  I don't feel super-strong about it, so feel free to leave it if you think it's better as-is.

Reply:
Done

Reply:
I didnt want to leave it as just "Use the credential" because then people might think they have to call GetToken themselves, whereas we do that for them in the SDK clients. So I added a message that expresses that, for the common case.